test/hotspot/jtreg/sources/SortIncludes.java

```diff
@@ -0,0 +1,198 @@
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+import java.util.SortedSet;
+import java.util.TreeSet;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+public class SortIncludes {
+    private static final String INCLUDE_LINE = "^ *#include *(<[^>]+>|\"[^\"]+\") *$\\n";
+    private static final String BLANK_LINE = "^$\\n";
+    private static final Pattern INCLUDES_RE = Pattern.compile(
+                    String.format("%s(?:(?:%s)*%s)*", INCLUDE_LINE, BLANK_LINE, INCLUDE_LINE),
+                    Pattern.MULTILINE);
+
+    /// Gets a sorting key for an include which is its substring as of `delim`, lowercased.
+    /// Note that using lowercase will sort `_` before letters.
+    private static Comparator<String> sortKeyForInclude(char delim) {
+        return Comparator.comparing(s -> s.toLowerCase().substring(s.indexOf(delim)));
+    }
+
+    /// Sorts the include statements in `block`.
+    ///
+    /// @param block source code chunk containing 1 or more include statements
+    /// @return `block` with the include statements sorted and a blank line between user and
+    /// sys includes
+    private static String sortedIncludes(String block) {
+        String[] lines = block.split("\\n");
+        SortedSet<String> userIncludes = new TreeSet<>(sortKeyForInclude('"'));
+        SortedSet<String> sysIncludes = new TreeSet<>(sortKeyForInclude('<'));
+        List<String> blankLines = new ArrayList<>();
+
+        // Partition lines into user include, sys includes and blank lines
+        for (String line : lines) {
+            if (line.contains("\"")) {
+                userIncludes.add(line);
+            } else if (line.contains("<")) {
+                sysIncludes.add(line);
+            } else if (line.isEmpty()) {
+                blankLines.add(line);
+            }
+        }
+
+        List<String> result = new ArrayList<>(userIncludes);
+        if (!userIncludes.isEmpty() && !sysIncludes.isEmpty() && blankLines.isEmpty()) {
+            blankLines = List.of("");
+        }
+        result.addAll(blankLines);
```

Comment:
If this line is removed you don't get the extra blank lines I mentioned in the previous PR. It also removes the extra blank line that you get inserted into oopMap.inline.hpp before the INCLUDE_JVMCI block.

Reply:
Or, rather if the code is changed to:
```
        if (!userIncludes.isEmpty() && !sysIncludes.isEmpty()) {
          result.add("");
        }
        result.addAll(sysIncludes);
```